### PR TITLE
Query API - 4 - added single queries with `first`

### PR DESCRIPTION
**Pull request by edualonso**
Blocked by https://github.com/realm/realm-kotlin/pull/609

Added single-element queries with `first`.
`asFlow` has been implemented using results, just as we did for scalar queries.

### CODE REVIEW

**Review by cmelchior — APPROVED** [review]
Just did a quick pass and didn't see anything problematic 👍

File: test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt [hunk 1]
```diff
@@ -1413,21 +1413,69 @@ class QueryTests {
...
+            .first()
+            .find { first ->
+                assertNotNull(first)
+                assertEquals(value2, first.intField)
+            }
     }
 
     @Test
-    @Ignore
     fun first_asFlow() {
```
Comment (rorbech) [inline test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt:1450]: What about the empty table case?

**Review by rorbech — APPROVED** [review]
LGTM ... except missing a test for `asFlow()` for empty tables.

File: test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt (hunk 1 above)
Comment (edualonso) [reply test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt:1608]: https://github.com/realm/realm-kotlin/blob/ce24690ff895881bed9a9827c12000f1b6ae7659/test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt#L1416

Comment (edualonso) [reply test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt:1608]: I missed the `asFlow` part. I have added it now.

**Comment by edualonso** [comment]
Added missing `deleteObservable` and `cancel` to the `asFlow` tests for `first`.

**Review by rorbech — APPROVED** [review]
